main.js

```diff
@@ -134,246 +78,271 @@ var getBlocks = () => {
+                let max = new BigNumber(balanceData.balance);
+
+                if (!fullBalances.includes(addr)) {
+                    max = max.times(config.payout);
+                }
+
+                balanceData.max = max.toNumber();
+
+                if (balanceData.balance > 1) {
+                    let share = utils.blockShareFunc(poolTotal, balanceData.max);
+                    balanceData.share = share[0];
 
                     //If they are blacklisted, keep their share
-                    if (blacklist[addr]) {
-                        balanceData[1] = new BigNumber(0);
+                    if (blacklist.includes(addr)) {
+                        balanceData.share = new BigNumber(0);
                         taxes = taxes.plus(share[0]);
                     }
 
                     taxes = taxes.plus(share[1]);
                     totalPay = totalPay.plus(share[0]);
                 }
+
             });
         });
 
-        console.log("TAXES: " + taxes.dividedBy(100000000));
-        console.log("PAYOUTS: " + totalPay.dividedBy(100000000));
         console.log("Voter weight calculation complete");
 
         resolve(taxes);
     })
 };
 
-var handleData = (taxes) => {
+let handleData = (taxes) => {
     return new Promise((resolve, rej) => {
-        var payouts = {};
-        for (let i = 0; i < sortedForgedBlocks.length; i++) {
-            var block = sortedForgedBlocks[i];
+        let payouts = {};
+
+        // Only pay up to number of blocks passed 
+        if (forgedToday > numBlocks)
+            forgedToday = numBlocks;
+
+        for (let i = sortedForgedBlocks.length - forgedToday; i < sortedForgedBlocks.length; i++) {
+            let block = sortedForgedBlocks[i];
             block.voterBalances.forEach((balanceData, addr) => {
-                var pay = payouts[addr] != null ? payouts[addr] : new BigNumber(0);
-                payouts[addr] = pay.plus(balanceData[1]);
+                let pay = payouts[addr] != null ? payouts[addr] : new BigNumber(0);
+                payouts[addr] = pay.plus(balanceData.share);
             });
-
-            /*
-            //For display purposes only
-            if (i == sortedForgedBlocks.length - 1) {
-                var asd = new BigNumber(0);
-                sortedForgedBlocks[i].voterBalances.forEach((balanceData, addr) => {
-                    asd = asd.plus(balanceData[1]);
-                });
-                console.log("last block sum: " + asd.dividedBy(100000000));
-            }
-            */
         }
-        var totalPayouts = new BigNumber(0);
+
+        let totalPayouts = new BigNumber(0);
         Object.keys(payouts).forEach((key) => totalPayouts = totalPayouts.plus(payouts[key]));
+        let taxes = new BigNumber(forgedToday * BLOCK_REWARD).minus(totalPayouts);
+        let ePayout = totalPayouts.dividedBy(100000000).dividedBy(new BigNumber(forgedToday).times(2)).times(100).toFormat(2);
```

Comment:
Extract the arktoshi<->ark conversion for bignums to a reusable function too. You can even have your conversion function from before handle both bignums and primitive ints.